service.go

```diff
@@ -61,26 +62,34 @@ func (s service) fetchResource(res factsetResource) error {
 	if err != nil {
 		return err
 	}
-	extension := filepath.Ext(res.fileName)
-	nameWithoutExt := strings.TrimSuffix(res.fileName, extension)
-	fileNameOnS3 := nameWithoutExt + "_" + fullVersion + extension
 
-	defer func() {
-		os.Remove(path.Join(dataFolder, fileName))
-		os.Remove(path.Join(dataFolder, fileNameOnS3))
-	}()
+	factsetFiles := strings.Split(res.fileNames, ";")
 
-	log.Infof("Resource [%s] was succesfully read from Factset in %d", fileName, time.Since(start))
+	for _, factsetFile := range factsetFiles {
+		extension := filepath.Ext(factsetFile)
+		nameWithoutExt := strings.TrimSuffix(factsetFile, extension)
+		fileNameOnS3 := nameWithoutExt + "_" + fullVersion + extension
+
+		log.Infof("Resource [%s] was succesfully read from Factset in %d", fileName, time.Since(start).String())
```

Comment:
Well it kicks off the download of all files as soon as its triggered, so yes. Im not really sure what you mean?